clickhouse_sqlalchemy/drivers/http/connector.py

```diff
@@ -126,24 +131,43 @@ def check_query_started(self):
     def fetchone(self):
         self.check_query_started()
 
-        if not self._rows:
-            return None
-
-        return self._rows.pop()
+        if self._rows is not None:
```

Comment:
If this for query streaming case checking for `self._prefetch` will be better here.

Reply:
It is, but in this way this particular logic doesn't have to know about prefetch, and can be re-used for many other cases (be it partial prefetch, optimized chunk-fetching, whatever). So I say this check is good as it is. A comment about prefetch would be better, prehaps?